### DISCUSSION TITLE
Eval Time Llama cpp vs Ollama. (80token/s. vs 800/token/s)

### DISCUSSION THREAD

**Discussion by michaellin99999** [ggml-org/llama.cpp#10495] [Q&A]
Why is there such a difference given that Ollama is a wrapper around Llamacpp. With the context legnth both set at 2048, I am able to run Ollama with eval time (prompt eval ) of 800ish tokens/s whereas LLamacpp only 80 per second.
Ollama is running on CPU (AMD Ryzen AI 9). and Llama cpp is running on iGPU. both on windows. is there something I'm missing?

**Comment by ExtReMLapin** [comment]
Could be because one is built for vulkan and the other not ?

**Reply by michaellin99999** [reply]
im able to build llamacpp with vulkan and frfom what i see vulkan is supported.     Ollama i have to hack it in. yet it is faster?
These are the settings i have for both:

Batch size 128, Context length 2048 Offload to iGPU 28 layers  KVcache is fp16.

**Reply by michaellin99999** [reply]
are there any settings that I am missing?
